lua/snacks/picker/core/picker.lua

```diff
@@ -415,19 +415,52 @@ function M:update_titles()
     if win.opts.title then
       local tpl = win.meta.title_tpl or win.opts.title
       win.meta.title_tpl = tpl
-      local ret = {} ---@type snacks.picker.Text[]
-      local title = Snacks.picker.util.tpl(tpl, data)
-      if title:find("{flags}", 1, true) then
-        title = title:gsub("{flags}", "")
-        vim.list_extend(ret, toggles)
-      end
-      title = vim.trim(title):gsub("%s+", " ")
-      if title ~= "" then
-        -- HACK: add extra space when last char is non word like an icon
-        title = title:sub(-1):match("[%w%p]") and title or title .. " "
-        table.insert(ret, 1, { " " .. title .. " ", "FloatTitle" })
+      if type(tpl) == "string" then
```

Comment:
instead of duplicating this, better to do
```lua
tpl = type(tpl) == "string" and {{tpl}} or tpl
```

Then use the same code to process

Reply:
that happens when one hasn't been writing lua in months
you forget about how to abuse ternary statements